src/ResourceManager/Network/Commands.Network/Cortex/HubVnetConnection/NewAzureRmHubVirtualNetworkConnectionCommand.cs

```diff
@@ -0,0 +1,176 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using AutoMapper;
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using System.Security;
+    using Microsoft.Azure.Commands.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+    using Microsoft.Azure.Management.Network;
+    using Microsoft.WindowsAzure.Commands.Common;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using System.Linq;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+    [Cmdlet(VerbsCommon.New,
+        "AzureRmHubVirtualNetworkConnection",
+        DefaultParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+        SupportsShouldProcess = true),
+        OutputType(typeof(PSHubVirtualNetworkConnection))]
+    public class NewHubVirtualNetworkConnectionCommand : HubVnetConnectionBaseCmdlet
+    {
+        [Alias("ResourceName", "HubVirtualNetworkConnectionName")]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource name.")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public virtual string ResourceGroupName { get; set; }
+
+        [Alias("VirtualHubName", "ParentVirtualHubName")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubName,
+            HelpMessage = "The parent resource name.")]
+        public string ParentResourceName { get; set; }
+
+        [Alias("VirtualHub", "ParentVirtualHub")]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CortexParameterSetNames.ByVirtualHubObject,
+            HelpMessage = "The parent resource.")]
+        public PSVirtualHub ParentResource { get; set; }
+
+        [Alias("VirtualHubId", "ParentVirtualHubId")]
+        [Parameter(
+            Mandatory = false,
```

Comment:
Any of these identity parameters in new/set/remove should always be mandatory.  Apply to all cmdlets.

Reply:
Fixed